src/Nethermind/Nethermind.TxPool/TxBroadcaster.cs

```diff
@@ -97,31 +97,31 @@ internal class TxBroadcaster : IDisposable
         // only for testing reasons
         internal Transaction[] GetSnapshot() => _persistentTxs.GetSnapshot();
 
-        public void Broadcast(Transaction tx, bool isPersistent)
+        public bool Broadcast(Transaction tx, bool isPersistent)
         {
             if (isPersistent)
             {
-                StartBroadcast(tx);
-            }
-            else
-            {
-                BroadcastOnce(tx);
+                return StartBroadcast(tx);
             }
+
+            BroadcastOnce(tx);
+
+            return true;
         }
 
-        private void StartBroadcast(Transaction tx)
+        private bool StartBroadcast(Transaction tx)
         {
             // broadcast local tx only if MaxFeePerGas is not lower than configurable percent of current base fee
             // (70% by default). Otherwise only add to persistent txs and broadcast when tx will be ready for inclusion
-            if (tx.MaxFeePerGas >= _baseFeeThreshold || tx.IsFree())
+            if (tx is not null
+                && _persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx)
+                && (tx.MaxFeePerGas >= _baseFeeThreshold || tx.IsFree()))
             {
                 NotifyPeersAboutLocalTx(tx);
+                return true;
             }
 
-            if (tx.Hash is not null)
-            {
-                _persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx);
-            }
+            return false;
         }
```

Comment:
We will reject now to add local txs if fee is lower than X percent (70% as default) of base fee, which is fine